File: backend/ŠišAppApi/Models/SalonAmenity.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace ŠišAppApi.Models;

public class SalonAmenity
{
    [Key]
    public int Id { get; set; }
    
    [Required]
    public int SalonId { get; set; }
    
    [ForeignKey("SalonId")]
    public Salon Salon { get; set; } = null!;
    
    [Required]
    [MaxLength(100)]
    public string Name { get; set; } = string.Empty;
    
    public string? Description { get; set; }
    
    public string? ImageId { get; set; }
    
    [ForeignKey("ImageId")]
    public Image? Image { get; set; }
    
    public bool IsAvailable { get; set; } = true;
    
    public int DisplayOrder { get; set; } = 0;
    
    [Required]
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    
    public DateTime? UpdatedAt { get; set; }
    
    public bool IsDeleted { get; set; } = false;
    
    public DateTime? DeletedAt { get; set; }
} 

File: sisapp_frontend/lib/models/salon_amenity.dart
class SalonAmenity {
  final int id;
  final int salonId;
  final String name;
  final String? description;
  final String? imageId;
  final bool isAvailable;
  final int displayOrder;

  SalonAmenity({
    required this.id,
    required this.salonId,
    required this.name,
    this.description,
    this.imageId,
    this.isAvailable = true,
    required this.displayOrder,
  });

  factory SalonAmenity.fromJson(Map<String, dynamic> json) {
    return SalonAmenity(
      id: (json['id'] as num?)?.toInt() ?? 0,
      salonId: (json['salonId'] as num?)?.toInt() ?? 0,
      name: json['name'] ?? '',
      description: json['description'],
      imageId: json['imageId'],
      isAvailable: json['isAvailable'] ?? true,
      displayOrder: (json['displayOrder'] as num?)?.toInt() ?? 0,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'salonId': salonId,
      'name': name,
      'description': description,
      'imageId': imageId,
      'isAvailable': isAvailable,
      'displayOrder': displayOrder,
    };
  }
}
